### PR TITLE
Enforce MedicalPersonnel to exist

### PR DESCRIPTION
**Pull request by diogenis-si**
## Description

issue #24 

A custom permission class takes place and validates if the MedicalPersonnel model exists.
## Checklist

- [ ] I have reviewed my code and checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ ] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation (**SwaggerHub**, **README.md**, **Wiki**, etc.) for the changes I have made

### CODE REVIEW

**Comment by dkoutsonikolis** [comment]
@diogenis-si please fix the build so we can review it, thanks!

**Comment by diogenis-si** [comment]
> @diogenis-si please fix the build so we can review it, thanks!

there is an error when isort parsing migration files, the error I get after running pre-commit locally thought the make file is the following:
`PermissionError: [Errno 13] Permission denied: 'tmh_registry/users/migrations/0007_auto_20210611_1126.py`

Have you faced something similar? Otherwise, I will have to do some search

P.S I also get permission errors when trying to modify migration files through Pycharm

**Comment by dkoutsonikolis** [comment]
> > @diogenis-si please fix the build so we can review it, thanks!
> 
> there is an error when isort parsing migration files, the error I get after running pre-commit locally thought the make file is the following:
> `PermissionError: [Errno 13] Permission denied: 'tmh_registry/users/migrations/0007_auto_20210611_1126.py`
> 
> Have you faced something similar? Otherwise, I will have to do some search
> 
> P.S I also get permission errors when trying to modify migration files through Pycharm

Yes, I have this issue too. When creating migrations through Docker, the owner of the file is the `root` account. So you have to change the owner to your user. My Linux user is called `dimitris` so I run this command `sudo chown dimitris:dimitris <migration_file>`. After that, everything works as intended.

**Comment by diogenis-si** [comment]
> > > @diogenis-si please fix the build so we can review it, thanks!
> > 
> > 
> > there is an error when isort parsing migration files, the error I get after running pre-commit locally thought the make file is the following:
> > `PermissionError: [Errno 13] Permission denied: 'tmh_registry/users/migrations/0007_auto_20210611_1126.py`
> > Have you faced something similar? Otherwise, I will have to do some search
> > P.S I also get permission errors when trying to modify migration files through Pycharm
> 
> Yes, I have this issue too. When creating migrations through Docker, the owner of the file is the `root` account. So you have to change the owner to your user. My Linux user is called `dimitris` so I run this command `sudo chown dimitris:dimitris <migration_file>`. After that, everything works as intended.

**Comment by diogenis-si** [comment]
@dkoutsonikolis build issues have been resovled after changing the migration file ownership and run pre-commit.

**Review by StefDoulfis — APPROVED** [review]

File: config/settings/base.py [hunk 1]
```diff
@@ -263,7 +263,7 @@
         "rest_framework.authentication.TokenAuthentication",
         "rest_framework.authentication.BasicAuthentication",
     ),
-    "DEFAULT_PERMISSION_CLASSES": ("rest_framework.permissions.IsAdminUser",),
+    "DEFAULT_PERMISSION_CLASSES": ("rest_framework.permissions.IsAdminUser", "tmh_registry.users.api.permissions.IsMedicalPersonnel"),
```
[the remark this answers is not in the document]
Comment (diogenis-si) [reply config/settings/base.py]: fixed!

File: tmh_registry/registry/tests/api/viewsets/test_hospitals.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 from rest_framework.test import APIClient
 
 from .....users.factories import MedicalPersonnelFactory
+from .....users.tests.factories import UserFactory
```
[the remark this answers is not in the document]
Comment (diogenis-si) [reply tmh_registry/registry/tests/api/viewsets/test_hospitals.py]: `user.factories` path used!

[the remark this answers is not in the document]
Comment (dkoutsonikolis) [reply tmh_registry/registry/tests/api/viewsets/test_hospitals.py]: Could you fix the path in the rest test files as well? Also, could you completely delete the `tmh_registry.users.tests.factories.py` file?

[the remark this answers is not in the document]
Comment (diogenis-si) [reply tmh_registry/registry/tests/api/viewsets/test_hospitals.py]: Can you please check again, I have deleted the file and so there are no remaining usages of `tmh_registry.users.tests.factories.py`

**Review by dkoutsonikolis — APPROVED** [review]

**Comment by dkoutsonikolis** [comment]
@diogenis-si thanks! If you have the time, please also update the Wiki https://github.com/swiftss-org/registry-api/wiki/Backend#users!

**Pull request merged**